### DISCUSSION TITLE
OOP SOLID Principles

### DISCUSSION THREAD

**Discussion by NinaKor130** [aai-institute/beyond-jupyter#7] [Ideas]
Suggestion.

Does it make sense actually to introduce SOLID principles in the OOP Essentials? It links pretty nicely with the example. For example, with the first step of extraction of the logic to the evaluation funciton you can explain how Single Responsibility Principle applies. I think it could be a nice guide and somehow a consistent rule of OOP.

**Comment by opcode81** [comment]
SOLID is part of the "OOP Essentials". It's just addressed after the example (case study). We relate the principles to the example later.

**Comment by NinaKor130** [comment]
Sorry, I expressed myself wrong, I meant introducing it earlier. I see that you introduce them in section 3. I feel like there is not enough importance given to it, and it would be nice to itnroduce them upfront and then with every example, ahve a specific section discussing how it is connected. You mention examples in section 3, but they "get lost". 

But this is just the taste.

**Reply by opcode81** [reply]
It was a conscious decision to do it the other way round (which we debated). Explaining the principles without any examples first is likely to make them appear too abstract; and if we provided examples for each of them separately, we would need more simplistic examples, which wouldn't be as meaningful.

**Reply by opcode81** [reply]
We are assuming low initial familiarity with OOP, yet SOLID can only be understood if you know OOP. Therefore we thought it was essential to have seen at least one example application first.

**Reply by NinaKor130** [reply]
Sounds good. Thank you for the explanation! I guess then the user feedback will be of crucial importance to check our assumptions!
